### ISSUE TITLE
removed unused variable and cause syntax error of bind function

### ISSUE THREAD

**Issue by loveppears** [rollup/rollup#3950]
- Rollup Version: 2.83.3
- Operating System (or Browser): Mac Chrome
- Node Version (if applicable): 14.53
- Link to reproduction (IMPORTANT, read below):  https://repl.it/@loveppears/rollup-repro-1

I am building a js library which requires a 3rd library `lottie`, and it has the following code

```
var wiggle = function wiggle(){}.bind(this)
```

But after I import it and build it, I got error in the line below

```
function wiggle(){}.bind(this)
```

uncaught syntaxerror unexpected identifier '.'

download the zip from `repl` and build it then open the `dist/index.html` , you could find this error.

Could you please tell me how can I avoid this

### Expected Behavior
```
var wiggle = function wiggle(){}.bind(this)
```
### Actual Behavior

```
function wiggle(){}.bind(this)
```

**Comment by kzc** [comment]
Same issue with object literals:
```
$ rollup -v
rollup v2.38.4

$ echo 'var x = {log: console.log}.log(123);' | rollup --silent | node
[stdin]:1
{log: console.log}.log(123);
                  ^
SyntaxError: Unexpected token '.'
```

**Comment by kzc** [comment]
and classes:
```
$ echo 'var a = class C{ static f(x){console.log(x)} }.f(123);' | rollup --silent | node
[stdin]:1
class C{ static f(x){console.log(x);} }.f(123);
                                       ^
SyntaxError: Unexpected token '.'
```

**Comment by lukastaegert** [comment]
Yes, I noticed. Working on a fix but might take a day.

**Comment by matthewp** [comment]
Introduced in this change: https://github.com/rollup/rollup/pull/3933

**Comment by lukastaegert** [comment]
Fix at #3951
